src/solidity/decodeInfo.js

```diff
@@ -0,0 +1,353 @@
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function FixedByteArray (type) {
+  if (type.split(' ')) {
+    type = type.split(' ')[0]
+  }
+  return {
+    needsFreeStorageSlot: false,
+    storageBytes: parseInt(type.replace('bytes', '')),
+    typeName: type.split(' ')[0],
+    name: 'fixedByteArray'
+  }
+}
+
+/**
+  * Int decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function Int (type) {
+  if (type.split(' ')) {
+    type = type.split(' ')[0]
+  }
+  return {
+    needsFreeStorageSlot: false,
+    storageBytes: parseInt(type.replace('int', '')) / 8,
+    typeName: type,
+    name: 'int'
+  }
+}
+
+/**
+  * StringType decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function StringType (type) {
+  return {
+    needsFreeStorageSlot: true,
+    storageBytes: 32,
+    typeName: 'string',
+    name: 'stringType'
+  }
+}
+
+/**
+  * ArrayType decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder, arraySize, subArray}
+  */
+function ArrayType (type, stateDefinitions) {
+  var arraySize
+  var storageBytes
+
+  var underlyingType = extractUnderlyingType(type)
+
+  arraySize = extractArraySize(type)
+  var dimensions = extractDimensions(type)
+  type = underlyingType + dimensions.join('')
+
+  var subArrayType = type.substring(0, type.lastIndexOf('['))
+  var subArray = null
+  if (subArrayType.indexOf('[') !== -1) {
+    subArray = decode(subArrayType, stateDefinitions)
+  }
+
+  underlyingType = decode(underlyingType, stateDefinitions)
+  storageBytes = underlyingType.storageBytes
+
+  if (arraySize === 'dynamic') {
+    storageBytes = 32
+  } else {
+    if (subArray) {
+      storageBytes = subArray.storageBytes // size on storage of one item of the array
+    }
+    storageBytes = Math.ceil(storageBytes * arraySize) // size on storage of one the whole array
+  }
+
+  return {
+    needsFreeStorageSlot: true,
+    storageBytes: storageBytes,
+    typeName: type,
+    arraySize: arraySize,
+    subArray: subArray,
+    name: 'arrayType'
+  }
+}
+
+/**
+  * Enum decode the given @arg type
+  *
+  * @param {String} type - type given by the AST
+  * @return {Object} returns decoded info about the current type: { needsFreeStorageSlot, storageBytes, typeName, decoder}
+  */
+function Enum (type, stateDefinitions) {
+  var extracted = type.split(' ')
+  return {
+    needsFreeStorageSlot: false,
+    storageBytes: 1,
```

Comment:
It is always at least 1 byte, but may be larger depending on the number of elements in the enum.